### PR TITLE
Publish releases on GitHub

### PR DESCRIPTION
**Pull request by adalinesimonian**
- Runs on every new push to main.
- Caches built dependencies between CI runs so as not to rebuild deps every time.
- Publishes builds as artifacts tied to the job.
- Builds for macOS (Apple Silicon and x86_64), Windows (x86_64), and Linux (MUSL on x86_64)
- Added link to latest build artifacts to readme

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+charset = utf-8
```
Comment (LFalch) [inline .editorconfig:7]: soft tabs of 4 spaces, utf-8 etc. are already the default for Rust. is there any reason to have it in the .editorconfig?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# talv
+
+Chess bot written in Rust. Comes with a simple GUI.
+
+## Download
```
Comment (LFalch) [inline README.md:5]: Maybe call it ## Download builds and also have a section for building it on your own

File: .editorconfig (hunk 1 above)
Comment (adalinesimonian) [reply .editorconfig:7]: convenient to ensure the IDE has the right settings

File: README.md (hunk 2 above)
Comment (adalinesimonian) [reply README.md:5]: sounds like a good idea! you should add that afterwards

**Pull request merged**